go/libraries/doltcore/diff/diffs.go

```diff
@@ -136,33 +136,33 @@ func (nd *DocDiffs) Len() int {
 }
 
 // GetDocDiffs retrieves staged and unstaged DocDiffs.
-func GetDocDiffs(ctx context.Context, dEnv *env.DoltEnv) (*DocDiffs, *DocDiffs, error) {
-	docDetails, err := dEnv.GetAllValidDocDetails()
+func GetDocDiffs(ctx context.Context, ddb *doltdb.DoltDB, rsr env.RepoStateReader) (*DocDiffs, *DocDiffs, error) {
+	docDetails, err := rsr.GetAllValidDocDetails()
 	if err != nil {
 		return nil, nil, err
 	}
 
-	workingRoot, err := dEnv.WorkingRoot(ctx)
+	workingRoot, err := env.WorkingRoot(ctx, ddb, rsr)
```

Comment:
Not crazy about these interfaces -- WorkingRoot etc. should be methods on RepoStateReader

Reply:
Agree. I will refactor this in a separate pr.